tasks/idrac_update.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+    - name: Check if dsu is installed
+      shell: command -v dsu
+      register: output_dsu_commad
+      failed_when: ( output_dsu_commad.stderr | length  > 0 )
+
+    - name: Set is_dsu_installed variable
+      set_fact:
+        is_dsu_installed: "{{ output_dsu_commad.stdout | length  > 0  }}"
+
+    - name: Upgrade the iDRAC if applicable 
```

Comment:
I also don't think you need to do that here, you can most likely add an option to dell_dsu_arguments that is conditional on upgrade_idrac_only

Reply:
I agree. I am updating the code to be generic and update any components instead only iDrac.